includes/options/class-amp-options-menu.php

```diff
@@ -1,54 +1,184 @@
+			array( $this, 'render_screen' ),
 			self::ICON_BASE64_SVG
 		);
 
+		add_submenu_page(
+			AMP_Options_Manager::OPTION_NAME,
+			__( 'AMP Settings', 'amp' ),
+			__( 'General', 'amp' ),
+			'manage_options',
+			AMP_Options_Manager::OPTION_NAME
+		);
+
+		add_settings_section(
+			'post_types',
+			false,
+			'__return_false',
+			AMP_Options_Manager::OPTION_NAME
+		);
+		add_settings_field(
+			'supported_post_types',
+			__( 'Post Type Support', 'amp' ),
+			array( $this, 'render_post_types_support' ),
+			AMP_Options_Manager::OPTION_NAME,
+			'post_types'
+		);
+
 		$submenus = array(
-			new AMP_Analytics_Options_Submenu( $this->menu_slug ),
+			new AMP_Analytics_Options_Submenu( AMP_Options_Manager::OPTION_NAME ),
 		);
 
 		// Create submenu items and calls on the Submenu Page object to render the actual contents of the page.
 		foreach ( $submenus as $submenu ) {
-			$submenu->init( $this->menu_slug );
+			$submenu->init();
 		}
-
-		$this->remove_toplevel_menu_item();
 	}
 
-	// Helper function to avoid having the top-level menu as
-	// the first menu item
-	function remove_toplevel_menu_item() {
-		global $submenu;
-		if ( isset( $submenu['amp-plugin-options'][0] ) ) {
-			unset( $submenu['amp-plugin-options'][0] );
+	/**
+	 * Check for errors with updating the supported post types.
+	 *
+	 * @since 0.6
+	 */
+	protected function check_supported_post_type_update_errors() {
+		$on_update = (
+			isset( $_GET['settings-updated'] ) // WPCS: CSRF ok.
+			&&
+			true === (bool) wp_unslash( $_GET['settings-updated'] ) // WPCS: CSRF ok.
+		);
+
+		// Only apply on update.
+		if ( ! $on_update ) {
+			return;
+		}
+
+		$builtin_support = AMP_Post_Type_Support::get_builtin_supported_post_types();
+		$settings        = AMP_Options_Manager::get_option( 'supported_post_types', array() );
+		foreach ( AMP_Post_Type_Support::get_eligible_post_types() as $name ) {
+			$post_type = get_post_type_object( $name );
+			if ( ! isset( $post_type->name, $post_type->label ) || in_array( $post_type->name, $builtin_support, true ) ) {
+				continue;
+			}
+
+			$post_type_support = post_type_supports( $post_type->name, AMP_QUERY_VAR );
+			$value             = ! empty( $settings[ $post_type->name ] );
+
+			$error = null;
+			if ( true === $value && true !== $post_type_support ) {
+				/* translators: %s: Post type name. */
+				$error = __( '"%s" could not be activated because support is removed by a plugin or theme', 'amp' );
+			} elseif ( empty( $value ) && true === $post_type_support ) {
+				/* translators: %s: Post type name. */
+				$error = __( '"%s" could not be deactivated because support is added by a plugin or theme', 'amp' );
+			}
+
+			if ( isset( $error ) ) {
+				add_settings_error(
+					$post_type->name,
+					$post_type->name,
+					sprintf(
+						$error,
+						$post_type->label
+					)
+				);
+			}
 		}
 	}
+
+	/**
+	 * Post types support section renderer.
+	 *
+	 * @since 0.6
+	 */
+	public function render_post_types_support() {
```

Comment:
Yeah, it's because we eventually want to move to Customizer and so adding another class temporarily for rendering HTML would seem undesirable if we're just going to delete it in the next release.